SSA/Projects/FullyHomomorphicEncryption/Basic.lean

```diff
@@ -607,15 +623,15 @@ theorem R.coeff_fromPoly {q n : Nat} [Fact (q > 1)] (p : (ZMod q)[X]) :
   3.
 -/
 /-- The coefficient of `fromTensor` is the same as the values available in the tensor input. -/
-theorem R.coeff_fromTensor [hqgt1 : Fact (q > 1)] (tensor : List Int)
+theorem R.coeff_fromTensor (tensor : List Int)
     (htensorlen : tensor.length < 2^n) :
     (R.fromTensor (q := q) (n := n) tensor).coeff i = (tensor.getD i 0) := by
   rw [fromTensor_eq_fromTensorFinsupp_fromPoly]
-  have hfromTensorFinsuppDegree := fromTensorFinsupp_degree q tensor
+  have hfromTensorFinsuppDegree := fromTensorFinsupp_degree (q:=q) tensor
```

Comment:
```suggestion
  have hfromTensorFinsuppDegree := fromTensorFinsupp_degree (q:=q) tensor
```

The fact that we need to supply the `q` explicitly is probably telling us to make it an explicit parameter. @goens I leave it to you as to whether you want to do this in the same PR. I'd be happy with a follow-up PR that makes the `q` explicit here!